app/client/components/user/createUser.jsx

```diff
@@ -1,422 +1,200 @@
+  }
+  if (window.IS_USING_BITCOIN && !values.blockchainAddress) {
+      errors.blockchainAddress = 'Missing Blockchain Address';
+  }
+  if (!window.IS_USING_BITCOIN && !values.publicSerialNumber) {
+      errors.publicSerialNumber = 'Missing Phone or ID';
+  }
+  return errors
 };
 
-const mapDispatchToProps = (dispatch) => {
-  return {
-    createUser: (body) => dispatch(createUser({body})),
-    resetCreateUser: () => {dispatch({type: RESET_CREATE_USER})},
-  };
-};
+let CreateUserForm = (props) => {
+    const { handleSubmit } = props;
+    let uniqueIdentifierInput;
+    let initialDisbursementAmount;
+
+    const renderField = ({
+      input,
+      label,
+      type,
+      children,
+      meta: { touched, error, warning }
+    }) => (
+        <div className={styles.inputBlock}>
+            <label className={styles.inputLabel}>{label}:</label>
+            <div style={{display: 'flex'}}>
+                <input className={styles.input} {...input} type={type} />
+                {children}
+            </div>
+            <div>
+                {touched &&
+                ((error && <span className={styles.error}>{error}</span>) ||
+                    (warning && <span>{warning}</span>))}
+            </div>
+        </div>
+    );
 
-class CreateUserForm extends React.Component {
+    if (!window.IS_USING_BITCOIN) {
+        uniqueIdentifierInput = (
+                <Field className={styles.input} name="publicSerialNumber" component={renderField} type="text" label={'Phone Number or ID'}>
+                    <QrReadingModal
+                        updateData={
+                            (data) => {
+                                //TODO (change this):
+                                //this.setState({
+                                //    publicSerialNumber: data.replace(/^\s+|\s+$/g, '')
+                                //})
+                            }
+                        }
+                    />
+                </Field>
+        )
+    } else {
+        uniqueIdentifierInput = <Field name="blockchainAddress" component={renderField} type="text" label='Bitcoin Address' />
+    }
 
-  constructor(props) {
-    super(props);
-    this.state = this.initialState;
+    if (window.MAXIMUM_CUSTOM_INITIAL_DISBURSEMENT > 0) {
+        initialDisbursementAmount = <Field name="customInitialDisbursement" component={renderField}
+                   type="text" label={'Initial Disbursement Amount'}>
+            {window.CURRENCY_NAME}
+        </Field>
+    }
 
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.handleCashierChange = this.handleCashierChange.bind(this);
-  }
+    return (
+        <div>
+            <ModuleHeader>Create a {props.transferAccountType} account</ModuleHeader>
+
+            <div style={{padding: '1em'}}>
+                <form onSubmit={handleSubmit}>
+                    <Field name="firstName" component={renderField} type="text" label={'First Name'} />
```

Comment:
before i just add it as `<Field component="input" className={styles.class} />` -- and just styled the field component itself. would something like that be possible in styled-components? styling custom components

Reply:
do u mean like this?
https://www.styled-components.com/docs/basics#styling-any-component